### PR TITLE
8288425: Footprint regression due MH creation when initializing StringConcatFactory

### PR DESCRIPTION
**Pull request by cl4es**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288425](https://bugs.openjdk.org/browse/JDK-8288425): Footprint regression due MH creation when initializing StringConcatFactory

### Reviewers
 * [Jim Laskey](https://openjdk.org/census#jlaskey) (@JimLaskey - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk19 pull/83/head:pull/83` \
`$ git checkout pull/83`

Update a local copy of the PR: \
`$ git checkout pull/83` \
`$ git pull https://git.openjdk.org/jdk19 pull/83/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 83`

View PR using the GUI difftool: \
`$ git pr show -t 83`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk19/pull/83.diff">https://git.openjdk.org/jdk19/pull/83.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java [hunk 1]
```diff
@@ -722,8 +722,15 @@ private static int classIndex(Class<?> cl) {
     private static final int[] PREPEND_FILTER_SECOND_PAIR_ARGS = new int[] { 0, 1, 4, 5 };
 
     // Base MH for complex prepender combinators.
-    private static final MethodHandle PREPEND_BASE = MethodHandles.dropArguments(
-            MethodHandles.identity(long.class), 1, byte[].class);
+    private static @Stable MethodHandle PREPEND_BASE;
```
Comment (JimLaskey) [inline src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java:725]: Left behind?

**Review by JimLaskey — APPROVED** [review]
LGTM

Is it possible to set up a regression test?

**Comment by cl4es** [comment]
> LGTM

Thanks. FWIW it's a clean backport. 

> Is it possible to set up a regression test?

We caught this automatically - albeit a bit later - using our automated startup and footprint tests. 

A functional pass-or-fail test is always a bit awkward when dealing with performance characteristics. We could consider some test that asserts that a Hello World doesn't spin up, say, any  `LambdaForm` classes dynamically, but there's nothing _wrong_ with using such things early. Future improvements could also mean we can do so for "free" (say by caching such classes in CDS or pre-generating everything in jlink).

**Pull request closed without merging**